packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -92,6 +99,11 @@ export interface IBucket extends IResource {
    */
   arnForObjects(keyPattern: string): string;
 
+  /**
+   * Grants write permissions for the LogDelivery group to the bucket.
+   */
+  allowLogDelivery(): void;
```

Comment:
I am still don't feel too good about this. The name is confusing, and users might think that this is how they can enable log delivery.

I am starting to think that maybe we should make this an internal API on `Bucket` (instead of `IBucket`). If people want to set the log delivery access control they can use the `accessControl` property when they create the bucket.

```ts
export class Bucket ... {
    /** @internal */
  public _allowLogDelivery() { ... }
};
```

Then, in the consuming side, you will need to check if your `IBucket` is a `Bucket`. Otherwise, you won't be able to use it for log delivery because you can't set it's access control:

```ts
if (props.serverAccessLogsBucket instanceof Bucket) {
  props.serverAccessLogsBucket._allowLogDelivery();
}
```

And that's it. I think it will simplify dramatically.

LMK if you want me to send you a PR with this proposal.

Reply:
Yea, sounds good. I moved the `allowLogDelivery` to the Bucket class and set it as a private method. I'm not sure what would be the difference by using the `@internal` annotation, tho.